code/datums/gas_mixture.dm

```diff
@@ -57,27 +57,27 @@ What are the archived variables for?
 /datum/gas_mixture/proc/heat_capacity()
 	var/heat_capacity = HEAT_CAPACITY_CALCULATION(oxygen,carbon_dioxide,nitrogen,toxins)
 
-	if(trace_gases.len)
-		for(var/datum/gas/trace_gas in trace_gases)

```

Comment:
So not only is there a redundant check for `trace_gases.len` up there, we also get to perform a redundant typecheck on every element of `trace_gases`, even though we KNOW that it will only ever contain gas datums.